src/contracts/types/ERC20.sol

```diff
@@ -0,0 +1,478 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```

Comment:
Curious about the file path here and the decision to name the folder types?

Also would recommend we stick with the 1 interface / contract / library per file paradigm and split this out further. I think it improves the code usability since no one has to dig to find where a library is nested in some file that isn't name the same.

Would recommend we do something like:

src/contracts
src/interfaces
src/libraries

Reply:
I split these out, idk why I put types inside contracts, I think it was a late night weekend coding thing.  I left the toke interfaces inside Staking.sol for now because later PR's will conflict with that.  I'll change that in the Claim PR because that's the last one I'll merge out of this LOW